File: codesignals/CheckPalindrome.java

package codesignals;
public class CheckPalindrome{

public static void main(String[] args) {
    String inputString = "radar";
    boolean isPalindrome = checkPalindrome(inputString);
    System.out.println("Is the string \"" + inputString + "\" a palindrome? " + isPalindrome);

}

public static boolean checkPalindrome(String inputString){
    int left =0;
    int right = inputString.length()-1;
    while(left< right){
        if(inputString.charAt(left)!=inputString.charAt(right)){
            return Boolean.FALSE;
        }
        left++;
        right--;
    }
    return Boolean.TRUE;
}
}

File: codesignals/CheckPalindrome.py
def CheckPalindrome(inputString):
    return inputString == inputString[::-1]

# Example usage:
if __name__ == "__main__":
    test_string = "radar"
    print(f"Is '{test_string}' a palindrome? {CheckPalindrome(test_string)}")
    
    test_string2 = "hello"
    print(f"Is '{test_string2}' a palindrome? {CheckPalindrome(test_string2)}")
    